modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -444,16 +444,22 @@ void reset(final Status st) {
             Closure oldDoneClosure = done;
 
             if (this.done != null) {
-                this.done = (Status status) -> {
+                Closure newDone = (Status status) -> {
                     if (status.isOk()) {
                         node.getOptions().getRaftGrpEvtsLsnr().onNewPeersConfigurationApplied(resultPeers);
                     } else {
-                        node.getOptions().getRaftGrpEvtsLsnr().onReconfigurationError(status);
+                        node.getOptions().getRaftGrpEvtsLsnr().onReconfigurationError(status, () -> {
+                            // TODO: error handling for changePeersAsync https://ggsystems.atlassian.net/browse/IGN-19294
```

Comment:
I'd rather expect such error handling to be on client side. There are few reasons for that, to mention some
- rebalance retry logic should be consistent with common sendWithRetry one. In case of inner handling (the one you've suggested above) it'll be possible to fail client side changePeersAsync with e.g. TimeoutException but still continue inner rebalance-retry logic. That seems to be invalid approach.
- It's incorrect to use metastore (for example in order to retrieve pending keys) from within NodeImpl.